wandb/sdk/wandb_init.py

```diff
@@ -174,7 +175,9 @@ def setup(self, kwargs: Any) -> None:  # noqa: C901
         # we add this logic to be backward compatible with the old behavior of disable
         # where it would disable the service if the mode was set to disabled
         mode = kwargs.get("mode")
-        settings_mode = (kwargs.get("settings") or {}).get("mode")
+        settings_mode = (kwargs.get("settings") or {}).get("mode") or os.environ.get(
+            "WANDB_MODE"
+        )
```

Comment:
Isn't this duplicating logic that exists in `Settings` that already reads the environment?

Reply:
we don't parse env var until after this point, so what was happening - we were still spinning up core if disabled via an env var...